File: cumulative-spring-boot-starter/src/test/java/org/excavator/boot/cumulative/test/CumulativeTests.java
/*
 * Licensed to the Apache Software Foundation (ASF) under one or more
 * contributor license agreements.  See the NOTICE file distributed with
 * this work for additional information regarding copyright ownership.
 * The ASF licenses this file to You under the Apache License, Version 2.0
 * (the "License"); you may not use this file except in compliance with
 * the License.  You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
package org.excavator.boot.cumulative.test;

import org.junit.jupiter.api.Test;
import static org.junit.jupiter.api.Assertions.*;
import org.junit.jupiter.api.extension.ExtendWith;
import org.springframework.boot.test.context.SpringBootTest;
import org.springframework.test.context.junit.jupiter.SpringExtension;

import javax.annotation.Resource;

@ExtendWith(SpringExtension.class)
@SpringBootTest(classes = CumulativeApplication.class)
public class CumulativeTests {

    @Resource
    CumulativeService cumulativeService;

    @Test
    public void testCumulativeByDay() {
        String key = "day";
        String dimensionKey = "testDay";
        String value = "10";
        String r = cumulativeService.cumulativeByDay(key, dimensionKey, value);

        assertTrue(Long.valueOf(r) >= Long.valueOf(value));
    }

    @Test
    public void testCumulativeByMonth() {
        String key = "month";
        String dimensionKey = "testMonth";
        String value = "10";
        String r = cumulativeService.cumulativeByMonth(key, dimensionKey, value);

        assertTrue(Long.valueOf(r) >= Long.valueOf(value));
    }

    @Test
    public void testCumulativeByDayAnMonth() {
        String key = "dayAndMonth";
        String dimensionKey = "testDayAndMonth";
        String value = "10";
        cumulativeService.cumulativeByDayAndMonth(key, dimensionKey, value);

        String dayValue = cumulativeService.queryByDay(key, dimensionKey);
        String monthValue = cumulativeService.queryByMonth(key, dimensionKey);

        assertTrue(Long.valueOf(dayValue) >= Long.valueOf(value));
        assertTrue(Long.valueOf(monthValue) >= Long.valueOf(value));
    }

    @Test
    public void testCumulativeByYear() {
        String key = "year";
        String dimensionKey = "testYear";
        String value = "10";
        String r = cumulativeService.cumulativeByYear(key, dimensionKey, value);

        assertTrue(Long.valueOf(r) >= Long.valueOf(value));

        r = cumulativeService.queryByYear(key, dimensionKey);

        assertTrue(Long.valueOf(r) >= Long.valueOf(value));
    }
}

File: cumulative-spring-boot-starter/src/test/scala/org/excavator/boot/cumulative/test/CumulativeTests.scala
package org.excavator.boot.cumulative.test

import javax.annotation.Resource
import org.junit.jupiter.api.Test
import org.junit.jupiter.api.Assertions._
import org.junit.jupiter.api.extension.ExtendWith
import org.springframework.boot.test.context.SpringBootTest
import org.springframework.test.context.junit.jupiter.SpringExtension

@ExtendWith(Array(classOf[SpringExtension]))
@SpringBootTest(classes = Array(classOf[CumulativeApplication]))
class CumulativeTests {
  @Resource
  private[test] val cumulativeService:CumulativeService = null

  @Test def testCumulativeByDay(): Unit = {
    val key = "day"
    val dimensionKey = "testDay"
    val value = "10"
    val r = cumulativeService.cumulativeByDay(key, dimensionKey, value)
    assertTrue(java.lang.Long.valueOf(r) >= java.lang.Long.valueOf(value))
  }

  @Test def testCumulativeByMonth(): Unit = {
    val key = "month"
    val dimensionKey = "testMonth"
    val value = "10"
    val r = cumulativeService.cumulativeByMonth(key, dimensionKey, value)
    assertTrue(java.lang.Long.valueOf(r) >= java.lang.Long.valueOf(value))
  }

  @Test def testCumulativeByDayAnMonth(): Unit = {
    val key = "dayAndMonth"
    val dimensionKey = "testDayAndMonth"
    val value = "10"
    cumulativeService.cumulativeByDayAndMonth(key, dimensionKey, value)
    val dayValue = cumulativeService.queryByDay(key, dimensionKey)
    val monthValue = cumulativeService.queryByMonth(key, dimensionKey)
    assertTrue(java.lang.Long.valueOf(dayValue) >= java.lang.Long.valueOf(value))
    assertTrue(java.lang.Long.valueOf(monthValue) >= java.lang.Long.valueOf(value))
  }

  @Test def testCumulativeByYear(): Unit = {
    val key = "year"
    val dimensionKey = "testYear"
    val value = "10"
    var r = cumulativeService.cumulativeByYear(key, dimensionKey, value)
    assertTrue(java.lang.Long.valueOf(r) >= java.lang.Long.valueOf(value))
    r = cumulativeService.queryByYear(key, dimensionKey)
    assertTrue(java.lang.Long.valueOf(r) >= java.lang.Long.valueOf(value))
  }
}
